### ISSUE TITLE
Cross-arch rebuild

### ISSUE THREAD

**Issue by artus9033** [electron/electron-rebuild#378]
I am using this library via electron-forge, but also tried it standalone, for armv7l cross-platform rebuild of native modules. Either way (electron-forge with armv7 arch configuration running on WSL, or through manual CLI execution with ```-a armv7l```), the resulting ```.node``` of the native modules that are being logged as rebuild targets, are compiled for x64, as ```readelf -a``` shows (Class: ELF64), both in the original node_modules, or in the copied .asar archive. Rebuild finishes without any errors. Am I doing something wrong or is cross-platform, cross-arch compilation not really supported?

**Comment by malept** [comment]
Cross compilation support is not currently supported, as [node-gyp doesn't currently support it](https://github.com/nodejs/node-gyp/issues/829). If you need this functionality, please add it to node-gyp first.

**Comment by artus9033** [comment]
Actually, while they claim it is not supported, after a while of experimenting, I got it working and successfully cross-compiled a bunch of native modules on WSL by installing ```arm-linux-gnueabihf-gcc``` & ```arm-linux-gnueabihf-g++```, and then specifying ```--arch=armv7``` and setting the environmental variables for the compilers: ```CC=arm-linux-gnueabihf-gcc CXX=arm-linux-gnueabihf-g++ CC_host="gcc -m32" CXX_host="g++ -m32" ...```, ensuring that the host compiler uses only 32-bit headers, so that even with more advanced modules there shouldn't be a problem.
Moreover, I actually got this working inline with invoking ```electron-forge```, so this solution isn't invasive in any way and the overhead to get this working is minimal. Maybe it should be possible for you to add simple config parameters, dependant on which the proper env variables would be set inline with the node-gyp invokation (can even be done host-platform-agnostically e.g. with [```cross-env```](https://www.npmjs.com/package/cross-env)
